app/controllers/articles_controller.rb

```diff
@@ -9,24 +9,24 @@ def create
         if @article.save
             redirect_to article_path(@article)
         else
-            render :new
+            render :new, status: :unprocessable_entity
         end
     end
 
     def show
-        @article = Article.find(params[:id])
+        @article = Article.find_by(slug: params[:slug])

```

Comment:
With this change, I'm curious if adding an index on the slug column would be a normal thing to do with rails (@fapapa?). I tend to do this when I'm creating the schema for the DB as I expect that this is the primary form of lookup on this record. 

Early on it won't matter since a sequential scan will still be fast. Eventually looking up an index will be needed as the table grows.